spec/syntax.md

```diff
@@ -148,19 +148,35 @@ MAY include an _annotation_ that is applied to the external value.
 
 A **_<dfn>local-declaration</dfn>_** binds a _variable_ to the resolved value of an _expression_.
 
-Declared _variables_ MUST NOT be used before their _declaration_,
-and their values MUST NOT be self-referential;
-otherwise, a _message_ is not considered _valid_.
-
-Multiple _declarations_ MUST NOT bind a value to the same _variable_;
-otherwise, a _message_ is not considered _valid_.
-
 ```abnf
 declaration = input-declaration / local-declaration
 input-declaration = input [s] variable-expression
 local-declaration = local s variable [s] "=" [s] expression
 ```
 
+_Variables_, once declared, MUST NOT be redeclared. 
+A _message_ that does any of the following is not _valid_ and will produce a 
+Duplicate Declaration error during formatting:
+- An _input-declaration_ MUST NOT bind a _variable_ that appears as a _variable_ in a previous 
+  _declaration_.
+- A _local-declaration_ MUST NOT bind a _variable_ that appears as a _variable_ in a previous
+  _declaration_.
```

Comment:
I thought local declarations were allowed to override external variables? Or was that a rejected proposal?

Reply:
It's a little complicated and I should probably add text about the corner case here.

Currently you can annotate an external value using `input`:
```
#input {$var :function opt=val}
```

... and you can assign it to a `local` variable:
```
#local $foo = {$var :function opt=val}
```

... and you can overwrite an external variable **_so long as it is not declared_**:

```
#local $var = {$foo :function opt=val}  // overwrites any `$var` passed in
```

But not this:

```
#input {$var :function opt=val}
#local $var = {$foo :something opt=val}
```